pkgs/by-name/sl/slimevr/default.nix

```diff
@@ -0,0 +1,221 @@
+    hash = "sha256-XQDbP+LO/brpl7viSxuV3H4ALN0yIkj9lwr5eS1txNs=";
+    # solarxr
+    fetchSubmodules = true;
+  };
+
+  sharedMeta = {
+    homepage = "https://docs.slimevr.dev/";
+    description = "App for facilitating full-body tracking in virtual reality";
+    license = with lib.licenses; [
+      mit
+      asl20
+    ];
+    maintainers = with lib.maintainers; [
+      gale-username
+      imurx
+    ];
+    platforms = with lib.platforms; darwin ++ linux;
+  };
+
+  # Without this the hidapi bundled with `org.hid4java:hid4java` will be used.
+  # The bundled version won't be able to find `libudev.so.1`.
+  serverJavaOptions =
+    let
+      # hid4java tries to load `libhidapi.so` which doesn't exist in Nix's hidapi.
+      # The `libhidapi.so` it expects is actually `libhidapi-hidraw.so`.
+      libhidapi = runCommand "libhidapi" { } ''
+        mkdir -p $out/lib
+        ln -s ${hidapi}/lib/libhidapi-hidraw.so $out/lib/libhidapi.so
+      '';
+    in
+    "-Djna.library.path='${
+      lib.makeLibraryPath [
+        hidapi
+        libhidapi
+      ]
+    }'";
+
+  server = stdenv.mkDerivation (finalAttrs: {
+    pname = "${pname}-server";
+    inherit version src;
+
+    mitmCache = gradle.fetchDeps {
+      inherit pname;
+      data = ./deps.json;
+    };
+
+    # Patch out use of grgit. Upstream code expects to be in a Git repo.
+    patches = [
+      ./no-grgit.patch
+    ];
+
+    postPatch = ''
+      # Manually apply program version, since Git can't be accessed.
+      substituteInPlace server/desktop/build.gradle.kts \
+        --replace-fail 'val gitVersionTag = ""' 'val gitVersionTag = "v${version}"'
+
+      # Disable Android, so its files don't have to be patched.
+      substituteInPlace settings.gradle.kts \
+        --replace-fail 'include(":server:android")' ""
+    '';
+
+    nativeBuildInputs = [
+      gradle
+      makeWrapper
+    ];
+
+    # this is required for using mitm-cache on Darwin
+    __darwinAllowLocalNetworking = true;
+
+    gradleFlags = [ "-Dorg.gradle.java.home=${jdk17_headless}" ];
+
+    gradleBuildTask = "shadowJar";
+
+    doCheck = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      install -Dm644 server/desktop/build/libs/slimevr.jar $out/share/slimevr/slimevr.jar
+      makeWrapper ${jdk17_headless}/bin/java $out/bin/slimevr-server \
+          --add-flags "${serverJavaOptions}" \
+          --add-flags "-jar $out/share/slimevr/slimevr.jar"
+
+      runHook postInstall
+    '';
+
+    meta = sharedMeta // {
+      mainProgram = "slimevr-server";
+    };
+  });
+in
+{
+  slimevr-server = server;
+  slimevr = rustPlatform.buildRustPackage {
+    inherit version pname src;
+
+    # `hyper-tls` and `rustls-pemfile` aren't in the upstream lockfile, but are in
+    # Tauri's. This lock file is the upstream one with both of those added.
+    # `hper-tls` and `rustls-pemfile` are optional dependencies of `reqwest`.
+    cargoLock = {
+      lockFile = ./Cargo.lock;
+    };
```

Comment:
yeah because the feature `native-tls`, native-tls-vendored` or `rustls-tls` is not enabled on the cargo file of SlimeVR when specifying it's tauri dependency....

why complicate the `cargoLock` via a local lock file when there is no usage of `tls` on compilation or runtime?

Reply:
didnt get reply yet

Reply:
i made a PR for the source https://github.com/SlimeVR/SlimeVR-Server/pull/1254

Reply:
Can't you delete the committed Cargo.lock file now?